pkg/rpc/settings.go

```diff
@@ -50,15 +50,15 @@ var useDialback = settings.RegisterBoolSetting(
 
 var enableRPCCompression = envutil.EnvOrDefaultBool("COCKROACH_ENABLE_RPC_COMPRESSION", true)
 
-func getWindowSize(name string, c ConnectionClass, defaultSize int) int32 {
+func getWindowSize(ctx context.Context, name string, c ConnectionClass, defaultSize int) int32 {
 	const maxWindowSize = defaultWindowSize * 32
 	s := envutil.EnvOrDefaultInt(name, defaultSize)
 	if s > maxWindowSize {
-		log.Warningf(context.Background(), "%s value too large; trimmed to %d", name, maxWindowSize)
+		log.Warningf(ctx, "%s value too large; trimmed to %d", name, maxWindowSize)
```

Comment:
Looking at the places where this is called (grpcDialRaw which is gossip and dial backs for example) it could become quite noisy in the logs. Should we limit this by `log.Every`? That's not on critical path and even when we make this configurable we would see it at least on first connection and then periodically for the case where log rolls.